enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,172 @@
+technology which can be used to distribute this information among Nodes in a
+cluster.
+
+There are other ways this control plane functionality can be implemented, such
+as using the Kubernetes API (OpenShift-SDN) or by using some other custom
+control plane technology (OVN-Kubernetes).
+
+Using BGP for this function is *not* a prerequisite for using BGP to
+satisfy other use cases.
+
+### External Service Load Balancing
+
+Kubernetes Services with `type=LoadBalancer` are typically used as an
+abstraction in front of a cloud's load balancer service.  Many on premise
+cluster environments do not have a load balancer service available, so these
+Services may not be used.
+
+One way to expose a Service is to set the `Service.spec.externalIPs` field.
+Once this field has been set, if the IP address(es) are routed to one or more
+Nodes in the cluster, traffic to the Service port(s) will be forwarded to the
+appropriate Service Endpoints.  The major catch with using this interface is
+that routing of these external IP addresses to the Nodes is left up as an
+exercise to the cluster administrator.
+
+Another way to implement `LoadBalancer` Services for these on premise
+clusters is to use BGP speaker on Nodes to publish routes to Nodes where
+`LoadBalancer` IP addresses may be reached.  [MetalLB][2] is one project that
+implements this technique.  [OpenShift Enhancement #356][3] discusses the use
+of MetalLB with OpenShift in more detail.
+
+### Exposing Pods or Services Directly
+
+Some users have expressed an interest in publishing routes for the Pod and
+Service networks via BGP.
+
+If we wanted to offer this, it would not be very difficult.  At least with
+OpenShift-SDN and OVN-Kubernetes, Nodes are already set up to handle traffic
+destined for Pod or Service cluster IP addresses if traffic were to arrive
+destined for those addresses.
+
+To enable this using BGP, a BGP speaker must run on each Node where we would
+like to serve as a router into the cluster network.  Each Node can publish a
+route for the Pod and Service network and upstream routers may use ECMP to
+distribute traffic among the Nodes.
+
+A future enhancement needs to be filed to pursue this integration in more
+detail.
+
+### Virtual Network Interconnect
+
+The Pod network is by default an isolated virtual network.  There is sometimes
+a desire to interconnect multiple virtual networks together.  This could be
+between multiple Kubernetes clusters, but it doesn't have to be the case.
+
+BGP EVPN (Ethernet VPN, or Virtual Private Network) is a BGP technology that is
+discussed in the context of this use case.  BGP EVPN can be used to advertise
+MAC addresses and IP-MAC address bindings.  BGP EVPN is not a full solution to
+this use case by itself, as it is a technology that can help implement part of
+the control plane.  There are more details that must be explored, including
+what the data plane looks like between the two clusters.
```

Comment:
Maybe we can expand and increase the scope of this section to Virtual Network Interconnect and Hybrid Cloud ?

Some thoughts below 
"BGP, specifically MP-BGP, is commonly used as a signaling protocol, together with other data planes technologies like MPLS and VXLAN to interconnect virtual network at L2 or L3"

BGP EVPN ..... (your description here)

BGP and MPLS is heavily used in the Telco area, and by the Cloud providers to interconnect their facilities to the Cloud, i.e. AWS Direct connect https://d1.awsstatic.com/whitepapers/Networking/integrating-aws-with-multiprotocol-label-switching.pdf
"
This makes it an attractive technology for hybrid cloud environments

and

Reply:
Thanks for the feedback!  I have incorporated your suggestions.

You also really left me hanging with "and" at the end!